### DISCUSSION TITLE
"Your curl version:" is missing

### DISCUSSION THREAD

**Discussion by ToshiEAB** [Homebrew/discussions#1994] [Linux]
Hello,

I've been trying to install Homebrew on my CentOS server. The server's default versions of cURL and Git were old and so I installed newer versions without deleting the old ones. Editing $PATH, the newer ones have become default versions and meet the requirements for Homebrew. Now I'm trying to install Homebrew with the following command:
```
sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)"
```
However, "Your curl version:" is missing and the installation terminates there as follows:

```
Press RETURN to continue or any other key to abort
==> /bin/chown -R serverID:members /home/serverID/.linuxbrew/Homebrew
==> Downloading and installing Homebrew...
HEAD is now at 90494ed8a Merge pull request #11852 from Homebrew/dependabot/bundler/Library/Homebrew/rubocop-1.19.0
/home/serverID/.linuxbrew/opt/curl/bin/curl: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
Error: Please update your system curl.
Minimum required version: 7.41.0
Your curl version:
Your curl executable: /home/serverID/opt/bin/curl
Failed during: /home/serverID/.linuxbrew/bin/brew update --force --quiet

[serverID@sv205 ~]$ which curl
~/opt/bin/curl

[serverID@sv205 ~]$ curl -V
curl 7.65.3 (x86_64-pc-linux-gnu) libcurl/7.65.3 OpenSSL/1.1.1b zlib/1.2.7
Release-Date: 2019-07-19
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS HTTPS-proxy Largefile libz NTLM NTLM_WB SSL TLS-SRP UnixSockets
```

The problem occurs only when installing with the "sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)"" command. I correctly see the curl version when running a portion of "brew.sh" program (i.e., I copied and pasted the portion to a separate .sh file). 

I tried many things suggested online like:
```
export HOMEBREW_DEVELOPER=1
export HOMEBREW_CURL_PATH="$HOME/opt/bin/curl"
```
or

`export HOMEBREW_FORCE_BREWED_CURL=1`
after making a copy of curl to $HOME/.linuxbrew/opt/curl/bin/

But none of them worked.

Can anyone help me figure out why the curl version is missing?

Thanks in advance :)

**Comment by danielnachun** [comment]
How did you install the newer version of `curl`?  The most reliable way to do this (which I have used on several CentOS installs now) is to install it with Miniconda, and then use 
```
export HOMEBREW_DEVELOPER=1
export HOMEBREW_CURL_PATH=(put your path to curl bin in Miniconda environment here)
export HOMEBREW_GIT_PATH=(put your path to git bin in Miniconda environment here)
```
You don't (and should not) need to activate the Miniconda environment to actually use the `curl` and `git` binaries installed in it, becuase they have RPATHs which will point them to the right libraries.  Once Homebrew is installed, you can get rid of the Anaconda environment as Homebrew will install it's own copies of `curl` and `git` that it uses.

**Reply by ToshiEAB** [reply]
@danielnachun

Thanks for your kind reply to my question. Following your instructions, I successfully installed Homebrew on CentOS. However, I am now faced with a different problem when trying the following command:

brew install gcc

Here is a portion of error messages: 
ld: cannot find crti.o: No such file or directory

The system does have crti.o: 

locate crti.o
/usr/lib64/crti.o

How can I let brew know the location of crti.o?

Thanks.

**Reply by kzhang2** [reply]
see here: https://github.com/Homebrew/discussions/discussions/1984
